src/index.d.ts

```diff
@@ -150,6 +150,9 @@ export declare function keys (obj: { [id: string]: any }): IterableIterator<stri
 export declare function values<T> (obj: { [id: string]: T }): IterableIterator<T>
 export declare function entries<T> (obj: { [id: string]: T }): IterableIterator<[string, T]>
 
+export declare function apply (fn: Function, args: Array<any>): any
+export declare function call (fn: Function, ...args: Array<any>): any
```

Comment:
I didn't, because typescript doesn't for `Function.prototype.call` or `Function.prototype.apply`. Tell me why it's not safe for them to use that type for the core function and I'll make this change.

Reply:
`Function.prototype.call` and `Function.prototype.apply` are methods, their argument types depend on argument types of the callee. Thus, in order to define argument types of `call` and `apply`, `interface Function` has to have generic parameter as well, this would case a breaking change, so TypeScript decided not to change it.

**TL;DR** `Function.prototype.call` and `Function.prototype.apply` doesn't do it not because of safety, but because of early design limitation.

Reply:
OK, great explanation, thank you! I wasn't thinking about that. I will change the definitions. @sithmel doesn't this seem like another good value-add for these functions?

Reply:
@KSXGitHub Do you happen to know what the lowest version of Typescript is that we support with our typedefs is? Typescript 2.8 has the `ReturnType` type, but I'm not sure if using that would reduce our support.

Reply:
3.0 since https://github.com/sithmel/iter-tools/pull/55. And I don't think this should concern us to be honest, very few (if none) uses old TypeScript versions.

Reply:
And BTW, why do you need `ReturnType`?

Reply:
Well the typedefs that you had me write seem to be part of what is now the `ReturnType` type, so I thought maybe we should just use that. I don't know why they do `infer R : R` though, and we don't have that bit.

Reply:
> Well the typedefs that you had me write seem to be part of what is now the `ReturnType` type

Oh, you were talking about that `Return` generic parameter? That is just a name, you can change it to anything (such as `R`, `T`) and it still works. I just chose `Return` because it is a return type.

Reply:
BTW, since don't know what built-it `ReturnType` is for (as you said "I don't know why they do `infer R : R` though"), I'm gonna explain it to you:

It is to infer return type of a function, for example:

```typescript
declare function myFunc (): 'foo' | 'bar'
type R = ReturnType<typeof myFunc> // R should be 'foo' | 'bar'
```

And `infer` is used to infer, for example:

```typescript
type ArrayElement<Arr> = Arr extends Array<infer X> ? X : any
type R = ArrayElement<[0, 1, 2, 3]> // R should be 0 | 1 | 2 | 3
```

Reply:
Typescript [says](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html):
"For example, the following extracts the return type of a function type:"
```ts
type ReturnType<T> = T extends (...args: any[]) => infer R ? R : any;
```

Reply:
I guess that's because the ReturnType type is used when you need to infer from another type. When you're extracting a generic type from a parameter it's obvious that you're "inferring".

Reply:
That's what I said